src/functionalTest/java/uk/gov/hmcts/divorce/testutil/FunctionalTestSuite.java

```diff
@@ -120,7 +120,7 @@ private CaseDetails submitNewCase(
     protected Response triggerCallback(Map<String, Object> caseData, String eventId, String url) throws IOException {
 
         // Add random Long generation of 16 digits to avoid duplicate case id
-        long caseId = Math.abs(new Random().nextLong());
+        long caseId = ThreadLocalRandom.current().nextLong(1_000_000_000_000_000L, 10_000_000_000_000_000L);
```

Comment:
Changed this because the F&P service request endpoint validates that the caseId is 16 digits long and the previous use of Random could generate IDs of different lengths

Reply:
no harm in having that explainer as comment really